pkg/vault/contracts/AddUnbalancedLiquidityViaSwapRouter.sol

```diff
@@ -98,13 +98,14 @@ contract AddUnbalancedLiquidityViaSwapRouter is RouterQueries, IAddUnbalancedLiq
         IERC20[] memory tokens;
         uint256[] memory amountsOut;
         (tokens, amountsIn, amountsOut) = _computeAddUnbalancedLiquidityViaSwap(hookParams);
-        for (uint256 i = 0; i < tokens.length; ++i) {
-            uint256 amountIn = amountsIn[i];
-            uint256 amountOut = amountsOut[i];
 
-            _takeTokenIn(hookParams.sender, tokens[i], amountIn, hookParams.wethIsEth);
-            _sendTokenOut(hookParams.sender, tokens[i], amountOut, hookParams.wethIsEth);
+        // In this case, the pools have two tokens,
+        // but we use a 'for' loop instead of getting the needed indexes because it's simpler.
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            _takeTokenIn(hookParams.sender, tokens[i], amountsIn[i], hookParams.wethIsEth);
+            _sendTokenOut(hookParams.sender, tokens[i], amountsOut[i], hookParams.wethIsEth);
```

Comment:
There's something weird about this conceptually (please let me know if I'm missing something).

The net result of this operation should only have tokens in:
- add proportional
- swap excess for exact (or vice versa depending on the outcome)

Either way, tokens should flow from the LP to the vault, and not vice versa. Why do we need to call `_sendTokenOut`? I'd seem to be unnecessary.